app/models/rbac.rb

```diff
@@ -459,12 +438,7 @@ def self.search(options = {})
   end
 
   def self.method_with_scope(ar_scope, options)
-    # for the most part, it is just asking if it extends ActsAsArModel
-    if ar_scope.try(:instances_are_derived?)
-      ar_scope.all(options)
-    else
-      ar_scope.apply_legacy_finder_options(options)
-    end
+    ar_scope.apply_legacy_finder_options(options)
```

Comment:
This is the most important change of the PR. Took over a year to make this change.

Since `ActsAsArModel` is just using scopes, we can directly call `apply_legacy_finder_options` (aka `where`, `limit`, `offset` ...)

Said another way, `options` no longer needs a single `:conditions` value. So sql munging will become a thing of the past.